streams/src/test/java/org/apache/kafka/streams/integration/KStreamRepartitionIntegrationTest.java

```diff
@@ -0,0 +1,859 @@
+
+        CLUSTER.createTopic(topicB, 1, 1);
+
+        final List<KeyValue<Integer, String>> expectedRecords = Arrays.asList(
+            new KeyValue<>(1, "A"),
+            new KeyValue<>(2, "B")
+        );
+
+        sendEvents(timestamp, expectedRecords);
+        sendEvents(topicB, timestamp, expectedRecords);
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final Repartitioned<Integer, String> inputTopicRepartitioned = Repartitioned
+            .<Integer, String>as(inputTopicRepartitionedName)
+            .withNumberOfPartitions(4);
+
+        final Repartitioned<Integer, String> topicBRepartitioned = Repartitioned
+            .<Integer, String>as(topicBRepartitionedName)
+            .withNumberOfPartitions(4);
+
+        final KStream<Integer, String> topicBStream = builder
+            .stream(topicB, Consumed.with(Serdes.Integer(), Serdes.String()))
+            .repartition(topicBRepartitioned);
+
+        builder.stream(inputTopic, Consumed.with(Serdes.Integer(), Serdes.String()))
+               .repartition(inputTopicRepartitioned)
+               .join(topicBStream, (value1, value2) -> value2, JoinWindows.of(Duration.ofSeconds(10)))
+               .to(outputTopic);
+
+        startStreams(builder);
+
+        assertEquals(4, getNumberOfPartitionsForTopic(toRepartitionTopicName(topicBRepartitionedName)));
+        assertEquals(4, getNumberOfPartitionsForTopic(toRepartitionTopicName(inputTopicRepartitionedName)));
+
+        validateReceivedMessages(
+            new IntegerDeserializer(),
+            new StringDeserializer(),
+            expectedRecords
+        );
+    }
+
+    @Test
+    public void shouldDoJoiningWhenNumberOfPartitionsOfRepartitionOperationMatchesNumberOfPartitionsOfSourceTopic() throws ExecutionException, InterruptedException {
+        final String topicBMapperName = "topic-b-mapper";
+        final String topicB = "topic-b-" + TEST_NUM.get();
+        final int topicBNumberOfPartitions = 6;
+        final String inputTopicRepartitionName = "join-repartition-test";
+
+        final long timestamp = System.currentTimeMillis();
+
+        CLUSTER.createTopic(topicB, topicBNumberOfPartitions, 1);
+
+        final List<KeyValue<Integer, String>> expectedRecords = Arrays.asList(
+            new KeyValue<>(1, "A"),
+            new KeyValue<>(2, "B")
+        );
+
+        sendEvents(inputTopic, timestamp, expectedRecords);
+        sendEvents(topicB, timestamp, expectedRecords);
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final Repartitioned<Integer, String> inputTopicRepartitioned = Repartitioned
+            .<Integer, String>as(inputTopicRepartitionName)
+            .withNumberOfPartitions(6);
+
+        final KStream<Integer, String> topicBStream = builder
+            .stream(topicB, Consumed.with(Serdes.Integer(), Serdes.String()))
+            .map(KeyValue::new, Named.as(topicBMapperName));
```

Comment:
Why do we have a `map()` step here? Wouldn't this imply a repartition topic that would match whatever number of partitions is used on the other stream? Ie, only without the map(), we guarantee that `topicBStream` has a certain number of partitions?

With the `map()` step it seems to be the same test `shouldDeductNumberOfPartitionsFromRepartitionOperation` as above?

Reply:
You're right. This test is redundant. Removed it.